lib/collection/src/shards/replica_set/shard_transfer.rs

```diff
@@ -145,10 +159,34 @@ impl ShardReplicaSet {
             _ => unreachable!(),
         };
 
-        let proxy_shard = QueueProxyShard::new(local_shard, remote_shard, wal_keep_from);
-        let _ = local.insert(Shard::QueueProxy(proxy_shard));
+        // Try to queue proxify with or without version
+        let proxy_shard = match from_version {
+            None => Ok(QueueProxyShard::new(
+                local_shard,
+                remote_shard,
+                wal_keep_from,
+            )),
+            Some(from_version) => QueueProxyShard::new_from_version(
+                local_shard,
+                remote_shard,
+                wal_keep_from,
+                from_version,
+            ),
+        };
 
-        Ok(())
+        // Insert queue proxy shard on success or revert to local shard on failure
+        match proxy_shard {
+            // All good, insert queue proxy shard
+            Ok(proxy_shard) => {
+                let _ = local.insert(Shard::QueueProxy(proxy_shard));
+                Ok(())
+            }
+            Err((local_shard, err)) => {
+                log::warn!("Failed to queue proxify shard, reverting to local shard: {err}");
+                let _ = local.insert(Shard::Local(local_shard));
+                Err(err)
```

Comment:
What will the caller of this function do in that case?

Reply:
That is up to the caller, but in our current code base we abort a shard transfer when this returns an error.

The important thing here is that reverting back to local brings it into the original state. That is required so the shard transfer (or something else) can be retried again.